plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestSqlServerCaseInsensitiveMapping.java

```diff
@@ -15,38 +15,70 @@
 
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
+import io.trino.plugin.jdbc.BaseCaseInsensitiveMappingTest;
 import io.trino.testing.QueryRunner;
+import io.trino.testing.sql.SqlExecutor;
 import org.testng.annotations.Test;
 
+import java.nio.file.Path;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.Statement;
+
+import static io.trino.plugin.jdbc.mapping.RuleBasedIdentifierMappingUtils.createRuleBasedIdentifierMappingFile;
 import static io.trino.plugin.sqlserver.SqlServerQueryRunner.createSqlServerQueryRunner;
 import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThat;
 
 // With case-insensitive-name-matching enabled colliding schema/table names are considered as errors.
 // Some tests here create colliding names which can cause any other concurrent test to fail.
 @Test(singleThreaded = true)
 public class TestSqlServerCaseInsensitiveMapping
-        extends BaseSqlServerCaseInsensitiveMappingTest
+        extends BaseCaseInsensitiveMappingTest
 {
+    private Path mappingFile;
     private TestingSqlServer sqlServer;
 
     @Override
     protected QueryRunner createQueryRunner()
             throws Exception
     {
+        mappingFile = createRuleBasedIdentifierMappingFile();
         sqlServer = closeAfterClass(new TestingSqlServer());
         sqlServer.start();
         return createSqlServerQueryRunner(
                 sqlServer,
                 ImmutableMap.of(),
                 ImmutableMap.<String, String>builder()
                         .put("case-insensitive-name-matching", "true")
+                        .put("case-insensitive-name-matching.config-file", mappingFile.toFile().getAbsolutePath())
+                        .put("case-insensitive-name-matching.config-file.refresh-period", "1ms") // ~always refresh
                         .build(),
                 ImmutableList.of());
     }
 
     @Override
-    protected TestingSqlServer getSqlServer()
+    protected Path getMappingFile()
+    {
+        return requireNonNull(mappingFile, "mappingFile is null");
+    }
+
+    @Override
+    protected SqlExecutor onRemote()
+    {
+        return requireNonNull(sqlServer, "sqlServer is null")::execute;
+    }
+
+    @Test
+    public void testSqlServerCollation()
```

Comment:
Separate commit.

Reply:
Also this seems more like a pre-condition? So maybe move it to test setup?

Reply:
The method was there already, see deleted BaseSqlServerCaseInsensitiveMappingTest